langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java

```diff
@@ -111,18 +117,31 @@ public void deployAsset() throws Exception {
             }
         }
 
+        private String getKeySpace() {
+            String keySpace =
+                    ConfigurationUtils.getString("keyspace", null, assetDefinition.getConfig());
+            return keySpace;
+        }
+
         @Override
-        public void deleteAsset() throws Exception {
-            throw new UnsupportedOperationException();
+        public boolean deleteAssetIfExists() throws Exception {
+            final String keySpace = getKeySpace();
+            List<String> statements =
+                    ConfigurationUtils.getList("delete-statements", assetDefinition.getConfig());
+            if (statements.isEmpty()) {
```

Comment:
I don't think this is a good idea, the table may have other dependencies like INDEXes it really depends on how you created it (and we have no control over the create-statements)

if there is no statement then we do nothing, like for create-statements

Reply:
it's not very intuitive. 
You set deletion-mode: delete and then you have to provide the keyspace delete statement ?
It's much easier if you just delete it. 

From my testing, the keyspace is being deleted even if not empty and it works well. 

Do you think an INDEX would make the keyspace deletion fail ?